core/embed/trezorhal/stm32u5/secret.c

```diff
@@ -143,6 +143,17 @@ secbool secret_optiga_present(void) {
   return secret_present(SECRET_OPTIGA_KEY_OFFSET, SECRET_OPTIGA_KEY_LEN);
 }
 
+secbool secret_optiga_set(const uint8_t *secret) {
+  uint8_t secret_enc[SECRET_OPTIGA_KEY_LEN] = {0};
+  if (sectrue != secure_aes_ecb_encrypt_hw(secret, sizeof(secret_enc),
+                                           secret_enc, SECURE_AES_KEY_DHUK)) {
+    return secfalse;
+  }
+  secret_write(secret_enc, SECRET_OPTIGA_KEY_OFFSET, SECRET_OPTIGA_KEY_LEN);
```

Comment:
Shouldn't we be calling  `secret_erase()` and `secret_write_header()` before `secret_write()` here?

Reply:
I actually removed it intentionally, sorry for not mentioning it. The idea is that secret sector contains also monotonic counter for bootloader downgrade protection, which i would rather not delete. At this point, the optiga pairing secret should be empty and writable, so no need to erase the sector. OTOH boardloader will reinitialze the monotonic counter on next run so probably no security issue if this gets deleted here.

Reply:
The fact that the logic around `secret_write_header()` is completely different between F4 and U5 is really not good. In U5 it's done by the bootloader and also the boardloader in `secret_ensure_initialized()`? Why both? In F4 it's done by prodtest in `secret_optiga_set()`. This is confusing and difficult to review.

> boardloader will reinitialze the monotonic counter on next run

Why? I don't see that it will do that.

Reply:
It seems to make sense to do this in the boardloader.
> but the F4 prodtest would still need to write the header too, to ensure compatibility with old boardloader

When we release a new version of prodtest, it will be released with a new boardloader and bundled in combined.bin. So I think we don't need to be concerned about that. I can't think of a scenario where we provisioned an F4 Trezor with an old combined.bin file, didn't run the Optiga pairing and then installed a newer prodtest version to run the Optiga pairing. Theoretically it can happen of course. On the other hand we will probably soon stop producing F4 devices, so if we are concerned about this scenario then, we can leave the code in prodtest temporarily until that moment comes.

Reply:
We have relatively large amount of devices that didnt pass test, some of these require reinstalling prodtest to do the retest, with no option to update the boardloader. If some of those were provided with new prodtest, and didn't do the pairing, it would be a problem. 

Yes i agree that these should all have optiga already paired, but i am not sure we can be entirely sure this does not happen

Reply:
Also. New boardloader, with old bootloader might be a problematic combination. Old bootloader, after downgrade via old firmware, will never see the optiga secret as wiped due to header being always present. This will prevent installing custom fw. Do we care though? Would need a bootloader monotonic version increment to prevent it.

Reply:
I think this is turning out to be out of scope of this PR. Let's discuss the problem IRL and resolve it in the other PR you mentioned elsewhere. For now I think we can remove the duplicate code from the bootloader.

Reply:
Removed: https://github.com/trezor/trezor-firmware/commit/c12fade159c80c2ddc6d81e5f15fe8d5d567cfdc